src/peakrdl_python/templates/addrmap.py.jinja

```diff
@@ -155,6 +163,33 @@ class {{get_fully_qualified_type_name(node)}}_cls(AddressMap):
         return self.__{{child_node.inst_name}}
             {%- endif %}
         {% endfor %}
+
+    {{ child_getter(node, "get_registers",   "Reg",        systemrdlRegNode) }}
+    {{ child_getter(node, "get_memories",    "Memory",     systemrdlMemNode) }}
+    {{ child_getter(node, "get_sections", "Union[AddressMap, RegFile]", (systemrdlAddrmapNode, systemrdlRegfileNode)) }}
+{%- endmacro %}
+
+{%- macro child_getter(node, getter_name, child_type, child_rdltype) %}
+    def {{getter_name}}(self, unroll=False) -> Iterable[Union[{{child_type}},Tuple[{{child_type}},...]]]:
```

Comment:
I think this should be type hinted as `Iterator ` rather than `Iterable` See[mypy type hints](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

Reply:
The python documentation for typing a generator, shows the use of `Iterator`. I have gone ahead and changed this.